### PR TITLE
Improve continuous integration + tweak to improve R CMD CHECK log

### PR DESCRIPTION
**Pull request by olivroy**
Addresses #774

Hi @Nowosad ,

here is a messy PR that rassembles my observations / modifications.

Adds CI GH actions (they fail, but I think it's fine. They can always be disabled if needed. But they really helped me fix some easy problems before the rest of the heavy lifting gets done

Note that the remotes field must be removed prior to CRAN submission. But leave it as long as these packages are not on CRAN (tmaptools and cols4all)

Edit: Some examples and tests are failing, this PR doesn't aim to fix that, but rather help pinpointing this

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -53,13 +55,23 @@ Development
...
+install_github("r-tmap/tmaptools")
+install_github("r-tmap/tmap")
+
+# On Linux, with pak
+# install.packages("pak")
+pak::pak("r-tmap/tmap")
+
+# Or from r-universe
+options(repos = c(
+  janmarvin = 'https://r-tmao.r-universe.dev',
```
Comment (Nowosad) [inline README.md:72]: ?

**Review by Nowosad — CHANGES_REQUESTED** [review]
@olivroy Great work! Please answer to my three comments...

@mtennekes Please take a look (and merge if you think everything is fine)

File: DESCRIPTION [hunk 2]
```diff
@@ -59,7 +59,11 @@ Suggests:
     covr,
     av,
     gifski
-URL: https://github.com/r-tmap/tmap
+URL: https://github.com/r-tmap/tmap, https://r-tmap.github.io/tmap/
+Remotes: r-tmap/tmaptools, mtennekes/cols4all
 BugReports: https://github.com/r-tmap/tmap/issues
 VignetteBuilder: knitr
 RoxygenNote: 7.2.3
+SystemRequirements: GDAL, sqlite3, v8, uduntis, GEOS, PROJ
```
[the remark this answers is not in the document]
Comment (olivroy) [reply DESCRIPTION:67]: well I don't know. It just that install seemed compkicated on Linux and Mac. This just tells R CmD Check to install those. and gives users information if they face issues. 

But I am really not a spatial specialist. I just know more about r packages in gemeral. 

And in tic or travis.yml, I saw mentions of those

File: R/step2_helper_data.R [hunk 3]
```diff
@@ -77,7 +77,9 @@ getdts = function(aes, unm, p, q, o, dt, shpvars, layer, mfun, args, plot.order)
 			check_fun = paste0("tmapValuesCheck_", unm)
 			if (!do.call(check_fun, list(x = val1))) {
 				# to do: add "layer" name e.g. tm_fill is still "polygons" and not "fill"
-				stop("Visual values used for the variable, \"", unm, "\" of layer function \"tm_", layer[1], "\" are incorrect.", call. = FALSE)
+				warning("Visual values used for the variable, \"", unm, "\" of layer function \"tm_", layer[1], "\" are incorrect.", call. = FALSE)
+				warning("This error was converted to a warning for testing.")
```
[the remark this answers is not in the document]
Comment (olivroy) [reply R/step2_helper_data.R:81]: Well. when I ran the tests I ran into this error. Converting it to a warning just helped see the other tests. 

But I will remove this

File: README.md (hunk 1 above)
Comment (olivroy) [reply README.md:72]: bad copypaste oops.

**Comment by olivroy** [comment]
@Nowosad Thanks you for your comments! Addressed the last 2

File: DESCRIPTION (hunk 2 above)
[the remark this answers is not in the document]
Comment (Nowosad) [reply DESCRIPTION:67]: @olivroy these are the requirements of packages such as sf and terra, etc. Another example is rmapshaper that requires v8, but this is done using the V8 package: thus it does not have systemrequirements but V8 has (see https://cran.r-project.org/package=rmapshaper). I do not know what is the CRAN policy about this (I was unable to find anything)

[the remark this answers is not in the document]
Comment (olivroy) [reply DESCRIPTION:67]: Ok I will remove. But if the r cmd chrck fails, we can always add them back

[the remark this answers is not in the document]
Comment (olivroy) [reply DESCRIPTION]: and note than Cran ignores config/needs

But the remotes field should be removed prior to Cran submission

**Comment by olivroy** [comment]
I also copied the content of NEWS from CRAN and changed it to NEWS.md to make it appear on the package website.

**Review by Nowosad — APPROVED** [review]

**Comment by Robinlovelace** [comment]
Looking good!

**Comment by olivroy** [comment]
Note: about the failed workflows notifications, that can be personalized in Github probably, for a repo and an organization. I think the notifications are not useful as there are too many things failing for now (possibly by design). 

But the logs are useful, to see where and when failures happen in the future.

**Comment by mtennekes** [comment]
Great team work, thx!

**Pull request merged**